src/parser.ts

```diff
@@ -781,19 +791,23 @@ export interface CustomConfigProviderItem {
     compilerArgs: string[];
     files: string[];
     windowsSDKVersion?: string;
+    currentPath: string;
+    line: string;
 }
 
 // Parse the output of the make dry-run command in order to provide CppTools
 // with information about includes, defines, compiler path....etc...
-// as needed by CustomConfigurationProvider
+// as needed by CustomConfigurationProvider. In addition generate a
+// CompileCommand entry for every file with a compiler invocation to build
+// a compile_commands.json file.
 export async function parseCustomConfigProvider(cancel: vscode.CancellationToken, dryRunOutputStr: string,
                                                 statusCallback: (message: string) => void,
                                                 onFoundCustomConfigProviderItem: (customConfigProviderItem: CustomConfigProviderItem) => void): Promise<number> {
     if (cancel.isCancellationRequested) {
         return make.ConfigureBuildReturnCodeTypes.cancelled;
     }
 
-    logger.message('Parsing dry-run output for CppTools Custom Configuration Provider.', "Normal");
+    logger.message('Parsing dry-run output for CppTools Custom Configuration Provider and compile_commands.json generation.', "Normal");
```

Comment:
I don't think we need to add this extra text here.

Reply:
Agree.